src/components/query-editor/EventsEditor.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { InlineFormLabel, Input, Select } from '@grafana/ui';
+import { SentryDataSource } from '../../datasource';
+import { selectors } from '../../selectors';
+import { SentryEventSortOptions } from '../../constants';
+import type { QueryEditorProps } from '@grafana/data';
+import type { SentryConfig, SentryQuery, SentryEventSort } from '../../types';
+
+type EventsEditorProps = Pick<QueryEditorProps<SentryDataSource, SentryQuery, SentryConfig>, 'query' | 'onChange' | 'onRunQuery'>;
+
+export const EventsEditor = ({ query, onChange, onRunQuery }: EventsEditorProps) => {
+  const onEventsQueryChange = (eventsQuery: string) => {
+    onChange({ ...query, eventsQuery } as SentryQuery);
+  };
+  const onEventsSortChange = (eventsSort: SentryEventSort) => {
+    onChange({ ...query, eventsSort: eventsSort || undefined } as SentryQuery);
+    onRunQuery();
+  };
+  const onEventsLimitChange = (eventsLimit?: number) => {
+    onChange({ ...query, eventsLimit: eventsLimit || undefined } as SentryQuery);
```

Comment:
```suggestion
    onChange({ ...query, eventsSort } as SentryQuery);
    onRunQuery();
  };
  const onEventsLimitChange = (eventsLimit?: number) => {
    onChange({ ...query, eventsLimit } as SentryQuery);
```

Nit: Do you need the `|| undefined` if it's already undefined and you're typecasting it to `SentryQuery`?